bypass censorship in Russia

matkoniecz opened the pull request:
provide fallback Overpass server (thanks to Kumi Systems for providing it, and thanks again to Roland Olbricht for providing main Overpass instance)

fixes #1389

-----

My work on this pull request was sponsored by a NGI Zero Discovery [grant](https://www.openstreetmap.org/user/Mateusz%20Konieczny/diary/368849)

matkoniecz commented on app/src/main/java/de/westnordost/streetcomplete/data/OsmModule.java:
```diff
@@ -62,18 +65,18 @@ public static OsmConnection osmConnection(OAuthConsumer consumer)
 	{
 		OsmConnection overpassConnection = new OsmConnection(
-				OVERPASS_API_URL, ApplicationConstants.USER_AGENT, null);
+			prefs.getString(Prefs.OVERPASS_URL, "https://overpass-api.de/api/"), ApplicationConstants.USER_AGENT, null);
```
I admit that after modifying this I was not expecting that everything will just work, as expected compilation failure.

So maybe I missed something else that should be done.

rugk replied:
```suggestion
			prefs.getString(Prefs.OVERPASS_URL, OVERPASS_API_URL), ApplicationConstants.USER_AGENT, null);
```

Same here? (I have no idea what I am suggesting, but well… :stuck_out_tongue_winking_eye: )

westnordost commented:
Aren't there any quests in StreetComplete that would fail to download if no attic data is available? (The construction quests)

matkoniecz commented:
The construction quests use attic data in test suite, not in normal operation.

Construction quests use "last modified" not requiring attic - but I may test is it actually working correctly.

westnordost commented:
Ah okay

mmd-osm commented on app/src/main/res/values/strings.xml:
```diff
@@ -649,10 +649,14 @@ Otherwise, you can download another keyboard in the app store. Popular keyboards
     <string name="label_housenumber">house number</string>
     <string name="label_blocknumber">block number</string>
     <string name="quest_busStopShelter_covered">Whole stop is covered</string>
+    <string name="overpass_server_default">Default overpass server provided by Roland Olbricht</string>
```
The server is provided/sponsored by FOSSGIS e.V.

matkoniecz replied:
What version of attribution would be best?

> Default overpass server provided by FOSSGIS, administered by Roland Olbricht

or 

> Default overpass server provided by FOSSGIS e.V., administered by Roland Olbricht

or

> Default overpass server provided by FOSSGIS e.V.


Pinging also @drolbr

matkoniecz replied:
moot as result of 

> Actually, I would very much prefer not to add a description of the Overpass URL to the localized strings but simply display the URL.

request

westnordost commented:
Actually, I would very much prefer not to add a description of the Overpass URL to the localized strings but simply display the URL.

On June 23, 2019 7:42:37 AM GMT+02:00, Mateusz Konieczny <notifications@github.com> wrote:
>matkoniecz commented on this pull request.
>
>
>
>>      <string
>name="quest_accessible_for_pedestrians_separate_sidewalk_explanation">This
>street was tagged as having no sidewalk on either side. In the case
>that there is a sidewalk after all but it is displayed as a separate
>way, please answer \"sidewalk\".</string>
><string name="action_open_location">Open location in another
>app</string>
><string name="map_application_missing">No other map application
>installed</string>
>     <string name="label_housenumber">house number</string>
>     <string name="label_blocknumber">block number</string>
><string name="quest_busStopShelter_covered">Whole stop is
>covered</string>
>+    <string name="overpass_server_default">Default overpass server
>provided by Roland Olbricht</string>
>
>What version of attribution would be best?
>
>> Default overpass server provided by FOSSGIS, administered by Roland
>Olbricht
>
>or 
>
>> Default overpass server provided by FOSSGIS e.V., administered by
>Roland Olbricht
>
>or
>
>> Default overpass server provided by FOSSGIS e.V.
>
>
>Pinging also @drolbr

matkoniecz commented:
> Actually, I would very much prefer not to add a description of the Overpass URL to the localized strings but simply display the URL.

Done.

matkoniecz commented:
> Construction quests use "last modified" not requiring attic - but I may test is it actually working correctly.

Tested to be sure. Attic queries are failing on Kumi servers (I reported this) but construction quest works without problems.

westnordost commented on app/src/main/java/de/westnordost/streetcomplete/data/OsmModule.java:
```diff
@@ -40,7 +43,7 @@
 	public static final String
 		OSM_API_URL = "https://api.openstreetmap.org/api/0.6/",
-		OVERPASS_API_URL = "https://overpass-api.de/api/",
+		OVERPASS_API_WITH_ATTIC_DATA_URL = "https://overpass-api.de/api/",
```
Why change the name of this constant?

matkoniecz replied:
Because OverpassOldMapDataDao actually need attic data to work (for running tests).

westnordost commented on app/src/main/java/de/westnordost/streetcomplete/data/OsmModule.java:
```diff
@@ -62,18 +65,18 @@ public static OsmConnection osmConnection(OAuthConsumer consumer)
 	{
 		OsmConnection overpassConnection = new OsmConnection(
-				OVERPASS_API_URL, ApplicationConstants.USER_AGENT, null);
+			prefs.getString(Prefs.OVERPASS_URL, "https://overpass-api.de/api/"), ApplicationConstants.USER_AGENT, null);
```
```suggestion
			prefs.getString(Prefs.OVERPASS_URL, OVERPASS_API_URL), ApplicationConstants.USER_AGENT, null);
```
?

rugk replied:
```suggestion
			prefs.getString(Prefs.OVERPASS_URL, OVERPASS_API_URL), ApplicationConstants.USER_AGENT, null);
```

direct suggestion :wink:

westnordost commented on app/src/main/res/values/arrays.xml:
```diff
@@ -24,4 +24,13 @@
+    <string-array name="pref_entries_overpass_url_select" translatable="false">
+        <item>"https://overpass-api.de/api/"</item>
+        <item>"https://overpass.kumi.systems/api/"</item>
+    </string-array>
```
Now this array is superfluous

matkoniecz replied:
Unfortunately, simply removing it fails. Both appear to be needed, otherwise build fails with errors like

```
Android resource linking failed
/home/mateusz/Documents/StreetComplete/app/src/main/res/xml/preferences.xml:107: error: resource array/pref_entryvalues_overpass_url_select (aka de.westnordost.streetcomplete.debug:array/pref_entryvalues_overpass_url_select) not found.
error: failed linking file resources.
```

```
org.gradle.api.tasks.TaskExecutionException: Execution failed for task ':app:processDebugResources'.
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:110)
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.execute(ExecuteActionsTaskExecuter.java:77)
	at org.gradle.api.internal.tasks.execution.OutputDirectoryCreatingTaskExecuter.execute(OutputDirectoryCreatingTaskExecuter.java:51)
	at org.gradle.api.internal.tasks.execution.SkipUpToDateTaskExecuter.execute(SkipUpToDateTaskExecuter.java:59)
	at org.gradle.api.internal.tasks.execution.ResolveTaskOutputCachingStateExecuter.execute(ResolveTaskOutputCachingStateExecuter.java:54)
	at org.gradle.api.internal.tasks.execution.ValidatingTaskExecuter.execute(ValidatingTaskExecuter.java:59)
	at org.gradle.api.internal.tasks.execution.SkipEmptySourceFilesTaskExecuter.execute(SkipEmptySourceFilesTaskExecuter.java:101)
	at org.gradle.api.internal.tasks.execution.FinalizeInputFilePropertiesTaskExecuter.execute(FinalizeInputFilePropertiesTaskExecuter.java:44)
	at org.gradle.api.internal.tasks.execution.CleanupStaleOutputsExecuter.execute(CleanupStaleOutputsExecuter.java:91)
	at org.gradle.api.internal.tasks.execution.ResolveTaskArtifactStateTaskExecuter.execute(ResolveTaskArtifactStateTaskExecuter.java:62)
	at org.gradle.api.internal.tasks.execution.SkipTaskWithNoActionsExecuter.execute(SkipTaskWithNoActionsExecuter.java:59)
	at org.gradle.api.internal.tasks.execution.SkipOnlyIfTaskExecuter.execute(SkipOnlyIfTaskExecuter.java:54)
	at org.gradle.api.internal.tasks.execution.ExecuteAtMostOnceTaskExecuter.execute(ExecuteAtMostOnceTaskExecuter.java:43)
	at org.gradle.api.internal.tasks.execution.CatchExceptionTaskExecuter.execute(CatchExceptionTaskExecuter.java:34)
	at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.run(EventFiringTaskExecuter.java:51)
	at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:300)
	at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:292)
	at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:174)
	at org.gradle.internal.operations.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:90)
	at org.gradle.internal.operations.DelegatingBuildOperationExecutor.run(DelegatingBuildOperationExecutor.java:31)
	at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter.execute(EventFiringTaskExecuter.java:46)
	at org.gradle.execution.taskgraph.LocalTaskInfoExecutor.execute(LocalTaskInfoExecutor.java:42)
	at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareWorkItemExecutor.execute(DefaultTaskExecutionGraph.java:277)
	at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareWorkItemExecutor.execute(DefaultTaskExecutionGraph.java:262)
	at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$ExecutorWorker$1.execute(DefaultTaskPlanExecutor.java:135)
	at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$ExecutorWorker$1.execute(DefaultTaskPlanExecutor.java:130)
	at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$ExecutorWorker.execute(DefaultTaskPlanExecutor.java:200)
	at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$ExecutorWorker.executeWithWork(DefaultTaskPlanExecutor.java:191)
	at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$ExecutorWorker.run(DefaultTaskPlanExecutor.java:130)
	at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor.process(DefaultTaskPlanExecutor.java:74)
	at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph.execute(DefaultTaskExecutionGraph.java:143)
	at org.gradle.execution.SelectedTaskExecutionAction.execute(SelectedTaskExecutionAction.java:40)
	at org.gradle.execution.DefaultBuildExecuter.execute(DefaultBuildExecuter.java:40)
	at org.gradle.execution.DefaultBuildExecuter.access$000(DefaultBuildExecuter.java:24)
	at org.gradle.execution.DefaultBuildExecuter$1.proceed(DefaultBuildExecuter.java:46)
	at org.gradle.execution.DryRunBuildExecutionAction.execute(DryRunBuildExecutionAction.java:49)
	at org.gradle.execution.DefaultBuildExecuter.execute(DefaultBuildExecuter.java:40)
	at org.gradle.execution.DefaultBuildExecuter.execute(DefaultBuildExecuter.java:33)
	at org.gradle.initialization.DefaultGradleLauncher$ExecuteTasks.run(DefaultGradleLauncher.java:355)
	at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:300)
	at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:292)
	at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:174)
	at org.gradle.internal.operations.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:90)
	at org.gradle.internal.operations.DelegatingBuildOperationExecutor.run(DelegatingBuildOperationExecutor.java:31)
	at org.gradle.initialization.DefaultGradleLauncher.runTasks(DefaultGradleLauncher.java:219)
	at org.gradle.initialization.DefaultGradleLauncher.doBuildStages(DefaultGradleLauncher.java:149)
	at org.gradle.initialization.DefaultGradleLauncher.executeTasks(DefaultGradleLauncher.java:124)
	at org.gradle.internal.invocation.GradleBuildController$1.call(GradleBuildController.java:77)
	at org.gradle.internal.invocation.GradleBuildController$1.call(GradleBuildController.java:74)
	at org.gradle.internal.work.DefaultWorkerLeaseService.withLocks(DefaultWorkerLeaseService.java:154)
	at org.gradle.internal.work.StopShieldingWorkerLeaseService.withLocks(StopShieldingWorkerLeaseService.java:38)
	at org.gradle.internal.invocation.GradleBuildController.doBuild(GradleBuildController.java:96)
	at org.gradle.internal.invocation.GradleBuildController.run(GradleBuildController.java:74)
	at org.gradle.tooling.internal.provider.runner.ClientProvidedBuildActionRunner.run(ClientProvidedBuildActionRunner.java:68)
	at org.gradle.launcher.exec.ChainingBuildActionRunner.run(ChainingBuildActionRunner.java:35)
	at org.gradle.launcher.exec.ChainingBuildActionRunner.run(ChainingBuildActionRunner.java:35)
	at org.gradle.tooling.internal.provider.ValidatingBuildActionRunner.run(ValidatingBuildActionRunner.java:32)
	at org.gradle.launcher.exec.RunAsBuildOperationBuildActionRunner$3.run(RunAsBuildOperationBuildActionRunner.java:50)
	at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:300)
	at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:292)
	at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:174)
	at org.gradle.internal.operations.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:90)
	at org.gradle.internal.operations.DelegatingBuildOperationExecutor.run(DelegatingBuildOperationExecutor.java:31)
	at org.gradle.launcher.exec.RunAsBuildOperationBuildActionRunner.run(RunAsBuildOperationBuildActionRunner.java:45)
	at org.gradle.tooling.internal.provider.SubscribableBuildActionRunner.run(SubscribableBuildActionRunner.java:51)
	at org.gradle.launcher.exec.InProcessBuildActionExecuter$1.transform(InProcessBuildActionExecuter.java:47)
	at org.gradle.launcher.exec.InProcessBuildActionExecuter$1.transform(InProcessBuildActionExecuter.java:44)
	at org.gradle.composite.internal.DefaultRootBuildState.run(DefaultRootBuildState.java:79)
	at org.gradle.launcher.exec.InProcessBuildActionExecuter.execute(InProcessBuildActionExecuter.java:44)
	at org.gradle.launcher.exec.InProcessBuildActionExecuter.execute(InProcessBuildActionExecuter.java:30)
	at org.gradle.launcher.exec.BuildTreeScopeBuildActionExecuter.execute(BuildTreeScopeBuildActionExecuter.java:39)
	at org.gradle.launcher.exec.BuildTreeScopeBuildActionExecuter.execute(BuildTreeScopeBuildActionExecuter.java:25)
	at org.gradle.tooling.internal.provider.ContinuousBuildActionExecuter.execute(ContinuousBuildActionExecuter.java:80)
	at org.gradle.tooling.internal.provider.ContinuousBuildActionExecuter.execute(ContinuousBuildActionExecuter.java:53)
	at org.gradle.tooling.internal.provider.ServicesSetupBuildActionExecuter.execute(ServicesSetupBuildActionExecuter.java:62)
	at org.gradle.tooling.internal.provider.ServicesSetupBuildActionExecuter.execute(ServicesSetupBuildActionExecuter.java:34)
	at org.gradle.tooling.internal.provider.GradleThreadBuildActionExecuter.execute(GradleThreadBuildActionExecuter.java:36)
	at org.gradle.tooling.internal.provider.GradleThreadBuildActionExecuter.execute(GradleThreadBuildActionExecuter.java:25)
	at org.gradle.tooling.internal.provider.ParallelismConfigurationBuildActionExecuter.execute(ParallelismConfigurationBuildActionExecuter.java:43)
	at org.gradle.tooling.internal.provider.ParallelismConfigurationBuildActionExecuter.execute(ParallelismConfigurationBuildActionExecuter.java:29)
	at org.gradle.tooling.internal.provider.StartParamsValidatingActionExecuter.execute(StartParamsValidatingActionExecuter.java:59)
	at org.gradle.tooling.internal.provider.StartParamsValidatingActionExecuter.execute(StartParamsValidatingActionExecuter.java:31)
	at org.gradle.tooling.internal.provider.SessionFailureReportingActionExecuter.execute(SessionFailureReportingActionExecuter.java:59)
	at org.gradle.tooling.internal.provider.SessionFailureReportingActionExecuter.execute(SessionFailureReportingActionExecuter.java:44)
	at org.gradle.tooling.internal.provider.SetupLoggingActionExecuter.execute(SetupLoggingActionExecuter.java:46)
	at org.gradle.tooling.internal.provider.SetupLoggingActionExecuter.execute(SetupLoggingActionExecuter.java:30)
	at org.gradle.launcher.daemon.server.exec.ExecuteBuild.doBuild(ExecuteBuild.java:67)
	at org.gradle.launcher.daemon.server.exec.BuildCommandOnly.execute(BuildCommandOnly.java:36)
	at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:122)
	at org.gradle.launcher.daemon.server.exec.WatchForDisconnection.execute(WatchForDisconnection.java:37)
	at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:122)
	at org.gradle.launcher.daemon.server.exec.ResetDeprecationLogger.execute(ResetDeprecationLogger.java:26)
	at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:122)
	at org.gradle.launcher.daemon.server.exec.RequestStopIfSingleUsedDaemon.execute(RequestStopIfSingleUsedDaemon.java:34)
	at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:122)
	at org.gradle.launcher.daemon.server.exec.ForwardClientInput$2.call(ForwardClientInput.java:74)
	at org.gradle.launcher.daemon.server.exec.ForwardClientInput$2.call(ForwardClientInput.java:72)
	at org.gradle.util.Swapper.swap(Swapper.java:38)
	at org.gradle.launcher.daemon.server.exec.ForwardClientInput.execute(ForwardClientInput.java:72)
	at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:122)
	at org.gradle.launcher.daemon.server.exec.LogAndCheckHealth.execute(LogAndCheckHealth.java:55)
	at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:122)
	at org.gradle.launcher.daemon.server.exec.LogToClient.doBuild(LogToClient.java:62)
	at org.gradle.launcher.daemon.server.exec.BuildCommandOnly.execute(BuildCommandOnly.java:36)
	at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:122)
	at org.gradle.launcher.daemon.server.exec.EstablishBuildEnvironment.doBuild(EstablishBuildEnvironment.java:81)
	at org.gradle.launcher.daemon.server.exec.BuildCommandOnly.execute(BuildCommandOnly.java:36)
	at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:122)
	at org.gradle.launcher.daemon.server.exec.StartBuildOrRespondWithBusy$1.run(StartBuildOrRespondWithBusy.java:50)
	at org.gradle.launcher.daemon.server.DaemonStateCoordinator$1.run(DaemonStateCoordinator.java:295)
	at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:63)
	at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:46)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:55)
	at java.lang.Thread.run(Thread.java:745)
Caused by: com.android.builder.internal.aapt.v2.Aapt2Exception: Android resource linking failed
/home/mateusz/Documents/StreetComplete/app/src/main/res/xml/preferences.xml:107: AAPT: error: resource array/pref_entryvalues_overpass_url_select (aka de.westnordost.streetcomplete.debug:array/pref_entryvalues_overpass_url_select) not found.
    
error: failed linking file resources.
	at com.android.builder.internal.aapt.v2.Aapt2Exception$Companion.create(Aapt2Exception.kt:45)
	at com.android.builder.internal.aapt.v2.Aapt2Exception$Companion.create$default(Aapt2Exception.kt:39)
	at com.android.build.gradle.internal.res.Aapt2ErrorUtils.rewriteException(Aapt2ErrorUtils.kt:97)
	at com.android.build.gradle.internal.res.Aapt2ErrorUtils.rewriteLinkException(Aapt2ErrorUtils.kt:73)
	at com.android.build.gradle.internal.res.LinkApplicationAndroidResourcesTask.invokeAaptForSplit(LinkApplicationAndroidResourcesTask.java:515)
	at com.android.build.gradle.internal.res.LinkApplicationAndroidResourcesTask.doFullTaskAction(LinkApplicationAndroidResourcesTask.java:257)
	at com.android.build.gradle.internal.tasks.IncrementalTask.taskAction(IncrementalTask.java:106)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.gradle.internal.reflect.JavaMethod.invoke(JavaMethod.java:73)
	at org.gradle.api.internal.project.taskfactory.IncrementalTaskAction.doExecute(IncrementalTaskAction.java:50)
	at org.gradle.api.internal.project.taskfactory.StandardTaskAction.execute(StandardTaskAction.java:39)
	at org.gradle.api.internal.project.taskfactory.StandardTaskAction.execute(StandardTaskAction.java:26)
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter$1.run(ExecuteActionsTaskExecuter.java:131)
	at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:300)
	at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:292)
	at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:174)
	at org.gradle.internal.operations.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:90)
	at org.gradle.internal.operations.DelegatingBuildOperationExecutor.run(DelegatingBuildOperationExecutor.java:31)
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeAction(ExecuteActionsTaskExecuter.java:120)
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:99)
	... 115 more
Caused by: com.android.builder.internal.aapt.v2.Aapt2Exception: Android resource linking failed
/home/mateusz/Documents/StreetComplete/app/src/main/res/xml/preferences.xml:107: error: resource array/pref_entryvalues_overpass_url_select (aka de.westnordost.streetcomplete.debug:array/pref_entryvalues_overpass_url_select) not found.
error: failed linking file resources.

	at com.android.builder.internal.aapt.v2.Aapt2Exception$Companion.create(Aapt2Exception.kt:45)
	at com.android.builder.internal.aapt.v2.Aapt2Exception$Companion.create$default(Aapt2Exception.kt:39)
	at com.android.builder.internal.aapt.v2.Aapt2DaemonImpl.doLink(Aapt2DaemonImpl.kt:188)
	at com.android.builder.internal.aapt.v2.Aapt2Daemon.link(Aapt2Daemon.kt:103)
	at com.android.builder.internal.aapt.v2.Aapt2DaemonManager$LeasedAaptDaemon.link(Aapt2DaemonManager.kt:176)
	at com.android.builder.core.AndroidBuilder.processResources(AndroidBuilder.java:839)
	at com.android.build.gradle.internal.res.LinkApplicationAndroidResourcesTask.invokeAaptForSplit(LinkApplicationAndroidResourcesTask.java:512)
	... 133 more
```

matkoniecz commented:
> at least not before an app restart

OK, that part was not tested - I changed setting, run queries that worked (using old server), added debug to confirm that server is changed, restarted, queries still worked and new server was appearing in logs.

And yes, restart is required for option to actually apply according to a proper testing.

> If there is no good solution, a cheap one would also be to require the user to re-start the app. Could be as cheap as mentioned in the description of the setting.

I did it for now. I will now try to research is there a good way to do this.

westnordost commented:
That's fine actually. We don't expect users to change this server all the time.

westnordost commented on app/src/main/java/de/westnordost/streetcomplete/data/OsmModule.java:
```diff
@@ -62,18 +65,18 @@ public static OsmConnection osmConnection(OAuthConsumer consumer)
 	{
 		OsmConnection overpassConnection = new OsmConnection(
-				OVERPASS_API_URL, ApplicationConstants.USER_AGENT, null);
+			prefs.getString(Prefs.OVERPASS_URL, OVERPASS_API_URL), ApplicationConstants.USER_AGENT, null);
```
How can this still compile?

matkoniecz replied:
It appears that I failed to push version that I actually tested.

matkoniecz replied:
Repushed, retested.

dbf256 commented:
thanks, waiting for next version :)

westnordost commented:
This will be the next major version, not the version I just released 10mins ago. Beta of that version will be out earliest in 2 weeks or so.

dbf256 commented:
Great, I see some complains (with 2 stars) from Russian users in play market, hope they will increase number of stars after this fix.

rugk commented:
> I see some complains (with 2 stars) from Russian users in play market

@westnordost should be able to reply to these complains. Maybe you can point them here… :smile:

matkoniecz commented:
> not the version I just released 10mins ago. Beta of that version will be out earliest in 2 weeks or so.

But texts for translations are already available for translation at POEditor (see https://github.com/westnordost/StreetComplete/blob/master/CONTRIBUTING.md#translating-the-app )
